### PR TITLE
Closes #590: Add exclude node title to page content type

### LINKED ISSUE #590: Add ability for content builders and editors to exclude node title
## Motivation
Just like in quickstart 1.0's flexible pages, the standard placement of node title can often look out of place on page content types

Most likely, the [exclude node title contrib module](http://drupal.org/project/exclude_node_title) is the path forward, mirroring functionality for QS1.0. And configuration could match QS1.0, where it's only applied to flexible page (or in QS2.0's case, page)

### Roles and Permissions considerations

- Anonymous users will either see or not see the node title, depending on editor/site builder decision 
- Content editor, Content administrator, and Administrators will have the option to hide the node title on page content type

### PR DESCRIPTION
**Pull request by barrettbaffert**
## Description

Just like in quickstart 1.0's flexible pages, the standard placement of node title can often look out of place on page content types

## Related Issue

https://github.com/az-digital/az_quickstart/issues/590

### How I tested locally with Lando. 

- Confirmed module is enabled on install
- Settings match QS1.0 here /admin/config/content/exclude-node-title
- Page content type has check mark for option to exclude node title
- Page with node title excluded still has an `<h1>` when you view source (an upgrade from QS1.0?)

### Questions

- Should settings be included in `config/install` folder, or within a module like `az_flexible_page/config/install`? 
- Which `.info` file should install the module, `az_quickstart.info.yml` or` az_flexible_page.info.yml`?
- Do we want pages with node title excluded to have a hidden `<h1>` when you view source? I assume yes, but want to be sure.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by barrettbaffert** [comment]
Node title excluded here: https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-597.probo.build/look-ma-no-node-title

**Comment by tadean** [comment]
Should this contain modifications to `config/install/user.role.az_content_admin.yml` or `config/install/user.role.az_content_editor.yml` to specify the exclude node title permissions?

**Comment by barrettbaffert** [comment]
> Should this contain modifications to `config/install/user.role.az_content_admin.yml` or `config/install/user.role.az_content_editor.yml` to specify the exclude node title permissions?

[Screenshot](https://arizona.box.com/s/5yr3xt2q1b7e5l3q0cxiaw7y8whzd7xk) of how permissions will be set (and I'll also need to apply config for user.role anonymous

**Comment by barrettbaffert** [comment]
Permissions can be confirmed here:

https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-597.probo.build/user/login?destination=admin/people/permissions#module-exclude_node_title

And I confirmed that this node title isn't visible in incognito browser

https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-597.probo.build/look-ma-no-node-title

**Review by tadean — APPROVED** [review]

File: config/install/user.role.anonymous.yml [hunk 1]
```diff
@@ -10,5 +10,6 @@ permissions:
   - 'access content'
   - 'access site-wide contact form'
   - 'search content'
+  - 'use exclude node title'
```
Comment (joeparsons) [inline config/install/user.role.anonymous.yml:13]: I think we may also need to add this `'use exclude node title'` permision to the authenticated user role.

**Review by joeparsons — APPROVED** [review]

**Comment by joeparsons** [comment]
Has merge conflicts in az_quickstart.info.yml (since we merged the blockquote PR)

**Comment by barrettbaffert** [comment]
> Has merge conflicts in az_quickstart.info.yml (since we merged the blockquote PR)

Thanks @joeparsons , should be resolved now!

**Review by smayden — APPROVED** [review]
This worked as expected for me when signed in as a content editor.

**Review by danahertzberg — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**